tests/BCL-Tests/Xamarin.Android.Bcl-Tests/Xamarin.Android.Bcl-Tests.targets

```diff
@@ -5,71 +5,89 @@
     <Exec
-        Command="$(RemapAssemblyRefTool) &quot;%(_Source.Identity)&quot; &quot;$(IntermediateOutputPath)%(_Source.Filename)%(_Source.Extension)&quot; nunitlite &quot;$(_NUnit)&quot;"
+        Condition=" '%(_TestAssembly.TestType)' == 'nunit' "
+        Command="$(RemapAssemblyRefTool) &quot;%(_TestAssembly.Identity)&quot; &quot;$(IntermediateOutputPath)%(_TestAssembly.RecursiveDir)%(_TestAssembly.Filename)%(_TestAssembly.Extension)&quot; nunitlite &quot;$(_NUnit)&quot;"
     />
-    <ItemGroup>
-      <_DebugSymbol
-          Include="..\..\..\bin\$(Configuration)\bcl-tests\%(MonoTestAssembly.Filename).pdb"
-          Condition="Exists('..\..\..\bin\$(Configuration)\bcl-tests\%(MonoTestAssembly.Filename).pdb')"
-      />
-    </ItemGroup>
     <Copy
-        SourceFiles="@(_DebugSymbol)"
-        DestinationFolder="$(IntermediateOutputPath)"
+        Condition=" '%(_TestAssembly.TestType)' == '' "
+        SourceFiles="@(_TestAssembly->'..\..\..\bin\$(Configuration)\bcl-tests\%(RecursiveDir)%(Filename)%(Extension)')"
+        DestinationFiles="@(_TestAssembly->'$(IntermediateOutputPath)%(RecursiveDir)%(Filename)%(Extension)')"
     />
-  </Target>
-  <ItemGroup>
-    <_TestResource Include="..\..\..\bin\$(Configuration)\bcl-tests\Test\resources\Resources.es-ES.resources">
-      <Culture>es-ES</Culture>
-      <Name>monodroid_corlib_test.resources.dll</Name>
-    </_TestResource>
-    <_TestResource Include="..\..\..\bin\$(Configuration)\bcl-tests\Test\resources\Resources.nn-NO.resources">
-      <Culture>nn-NO</Culture>
-      <Name>monodroid_corlib_test.resources.dll</Name>
-    </_TestResource>
-  </ItemGroup>
-  <Target Name="_CreateTestResources"
-      BeforeTargets="ResolveAssemblyReferences"
-      DependsOnTargets="_ExtractTestResources"
-      Inputs="@(_TestResource)"
-      Outputs="$(IntermediateOutputPath)%(_TestResource.Culture)\%(_TestResource.Name)">
-    <MakeDir Directories="$(IntermediateOutputPath)%(_TestResource.Culture)" />
-    <AL
-        OutputAssembly="$(IntermediateOutputPath)%(Culture)\%(Name)"
-        Version="1.0.0.0"
-        Culture="%(Culture)"
-        EmbedResources="@(_TestResource)"
+    <Copy
+        Condition="Exists('..\..\..\bin\$(Configuration)\bcl-tests\%(_TestAssembly.RecursiveDir)%(_TestAssembly.Filename).pdb')"
+        SourceFiles="@(_TestAssembly->'..\..\..\bin\$(Configuration)\bcl-tests\%(RecursiveDir)%(Filename).pdb')"
+        DestinationFiles="@(_TestAssembly->'$(IntermediateOutputPath)%(RecursiveDir)%(Filename).pdb')"
     />
   </Target>
-  <Target Name="_ExtractTestResources"
-      Inputs="..\..\..\bin\$(Configuration)\bcl-tests\bcl-tests.zip"
-      Outputs="@(_TestResource)">
-    <SystemUnzip
-        SourceFiles="..\..\..\bin\$(Configuration)\bcl-tests\bcl-tests.zip"
-        HostOS="$(HostOS)"
-        DestinationFolder="..\..\..\bin\$(Configuration)\bcl-tests"
+  <Target Name="_WriteAssembliesList"
+      Outputs="nunit-assemblies.txt;xunit-assemblies.txt"
```

Comment:
There's no point in having an `Outputs` unless you also have an `Inputs`.

What could `Inputs` even be?  the `bcl-tests.zip` file?

Additionally, if you *do* add a `Inputs`, normal/expected MSBuild semantics are that *if* the `Inputs` change, then `Outputs` *must* be touched.  The use of `WriteOnlyWhenDifferent` (below) would break this assumption.